arraycontext/container/traversal.py

```diff
@@ -583,7 +583,10 @@ def thaw(ary: ArrayOrContainerT, actx: ArrayContext) -> ArrayOrContainerT:
 
 # {{{ flatten / unflatten
 
-def flatten(ary: ArrayOrContainerT, actx: ArrayContext) -> Any:
+def flatten(
+        ary: ArrayOrContainerT, actx: ArrayContext, *,
+        leaf_class: Optional[type] = None,
```

Comment:
Document `leaf_class`.

Reply:
Added some docs in 5e87623.

Should also do the same for all the other functions using `leaf_class`, since none of them had some nice docs I could copy-paste :cry: